sdk/servicebus/Azure.Messaging.ServiceBus/samples/Sample15_MockingClientTypes.md

```diff
@@ -0,0 +1,619 @@
+
+BinaryData setState = new("MockState");
+await sessionReceiver.SetSessionStateAsync(setState, CancellationToken.None);
+BinaryData state = await sessionReceiver.GetSessionStateAsync(CancellationToken.None);
+
+Assert.AreEqual(setState, state);
+```
+
+## Sending and Receiving messages using topics and subscriptions
+
+```C# Snippet:ServiceBus_MockingTopicSubscriptionSend
+Mock<ServiceBusAdministrationClient> mockAdministrationClient = new();
+Mock<Response<TopicProperties>> mockTopicResponse = new();
+Mock<Response<SubscriptionProperties>> mockSuscriptionResponse = new();
+
+mockAdministrationClient
+    .Setup(client => client.CreateTopicAsync(
+        It.IsAny<CreateTopicOptions>(),
+        It.IsAny<CancellationToken>()))
+    .Callback<CreateTopicOptions, CancellationToken>((opts, ct) =>
+    {
+        TopicProperties mockTopicProperties = ServiceBusModelFactory.TopicProperties(opts);
+
+        mockTopicResponse.Setup(r => r.Value).Returns(mockTopicProperties);
+    })
+    .ReturnsAsync(mockTopicResponse.Object);
+
+mockAdministrationClient
+    .Setup(client => client.CreateSubscriptionAsync(
+        It.IsAny<CreateSubscriptionOptions>(),
+        It.IsAny<CancellationToken>()))
+    .Callback<CreateSubscriptionOptions, CancellationToken>((opts, ct) =>
+    {
+        SubscriptionProperties mockSubscriptionProperties = ServiceBusModelFactory.SubscriptionProperties(opts);
+
+        mockSuscriptionResponse.Setup(r => r.Value).Returns(mockSubscriptionProperties);
+    })
+    .ReturnsAsync(mockSuscriptionResponse.Object);
+
+var adminClient = mockAdministrationClient.Object;
+
+Mock<ServiceBusClient> mockServiceBusClient = new();
+Mock<ServiceBusSender> mockSender = new();
+Mock<ServiceBusReceiver> mockReceiver = new();
+
+mockServiceBusClient
+    .Setup(client => client.CreateSender(It.IsAny<string>()))
+    .Returns(mockSender.Object);
+
+mockSender
+    .Setup(sender => sender.SendMessageAsync(
+        It.IsAny<ServiceBusMessage>(),
+        It.IsAny<CancellationToken>()))
+    .Returns(Task.CompletedTask);
+
+mockServiceBusClient
+    .Setup(client => client.CreateReceiver(
+        It.IsAny<string>(),
+        It.IsAny<string>()))
+    .Returns(mockReceiver.Object);
+
+async IAsyncEnumerable<ServiceBusReceivedMessage> mockReturn()
+{
+    ServiceBusReceivedMessage message = ServiceBusModelFactory.ServiceBusReceivedMessage(
+        body: new BinaryData("message1"),
+        messageId: "messageId1",
+        partitionKey: "hellokey",
+        correlationId: "correlationId",
+        contentType: "contentType",
+        replyTo: "replyTo"
+        // see the Service Bus Model Factory for more options ...
+        );
+
+    // IAsyncEnumerable types can only be returned by async functions, use this no-op await statement to
+    // force the method to be async.
+
+    await Task.Yield();
+
+    // yield allows more than one message to be returned, only one is being returned here for brevity
```

Comment:
This comment may be confusing; `yield` allows a single item to be emitted at a time.   The comment could be interpreted as `yield` being able to return multiple items from a single call.